src/ore/src/now.rs

```diff
@@ -86,6 +86,30 @@ pub static SYSTEM_TIME: Lazy<NowFn> = Lazy::new(|| NowFn::from(system_time));
 /// For use in tests.
 pub static NOW_ZERO: Lazy<NowFn> = Lazy::new(|| NowFn::from(now_zero));
 
+/// Provides means to make a [`NowFn`] stateful and ensure values are strictly
+/// monotonic.
+#[derive(Debug, Clone)]
+pub struct MonotonicNow {
+    max_ts: EpochMillis,
+    now: NowFn,
+}
+
+impl MonotonicNow {
+    /// Return a monotonic `EpochMillis` value.
+    ///
+    /// Note that the minimum value this function ever returns is `1`.
+    pub fn now(&mut self) -> EpochMillis {
```

Comment:
I removed `MonotonicNow` in favor of having the `PersistWriteHandle` offer a monotonic append that chooses all of the timestamps. Has a few advantages:
- We don't need to add retry logic for our writes bc currently we know we're the only writer to these collections.
- We don't need to track the maximum upper when adding new collections to avoid adding retry logic.